web/src/timerender.ts

```diff
@@ -333,10 +336,10 @@ function render_date_span($elem: JQuery, rendered_time: TimeRender): JQuery {
 // (What's actually spliced into the message template is the contents
 // of this DOM node as HTML, so effectively a copy of the node. That's
 // okay since to update the time later we look up the node by its id.)
-export function render_date(time: Date, today: Date): JQuery {
+export function render_date(time: Date, today: Date, prev_elem_time?: Date): JQuery {
     const className = `timerender${next_timerender_id}`;
     next_timerender_id += 1;
-    const rendered_time = render_now(time, today);
+    const rendered_time = render_now(time, today, prev_elem_time);
```

Comment:
Let's move the changes in this file in a separate commit and add tests for it. These function effect more than just the message header.